standards/ICRC-3/README.md

```diff
@@ -86,238 +86,661 @@ An ICRC-3 compliant Block
         }};
-    }}};
-}};
 ```
 
-Example without `btype`:
+
+Example of account representation as an array with one blob encoding the owner principal.
 ```
-variant { Map = vec {
-    record { "phash"; variant {
-        Blob = blob "\a1\a9p\f5\17\e5\e2\92\87\96(\c8\f1\88iM\0d(tN\f4-~u\19\88\83\d8_\b2\01\ec"
-    }};
-    record { "ts"; variant { Nat = 1_701_108_969_851_098_255 : nat }};
-    record { "tx"; variant { Map = vec {
-        record { "op"; variant { Text = "burn" } };
-        record { "amt"; variant { Nat = 1_228_990 : nat } };
-        record { "from"; variant { Array = vec {
-                variant { Blob = blob "\00\00\00\00\020\00\07\01\01" };
-                variant { Blob = blob "&\99\c0H\7f\a4\a5Q\af\c7\f4;\d9\e9\ca\e5 \e3\94\84\b5c\b6\97/\00\e6\a0\e9\d3p\1a" };
-        }}};
-        record { "memo"; variant { Blob = blob "\82\00\83x\223K7Bg3LUkiXZ5hatPT1b9h3XxJ89DYSU2e\19\07\d0\00"
+variant { Array = vec {
+                variant { Blob = blob "\00\00\00\00\00\f0\13x\01\01" };
+            
         }};
-    }}};
-}};
 ```
 
-#### Mint Block Schema
 
-1. the `btype` field MUST be `"1mint"` or the `tx.op` field MUST be `"mint"`
-2. it MUST contain a field `tx.to: Account`
+### Canonical `tx` Mapping
+
+Each ICRC-1 or ICRC-2 method call maps deterministically to the `tx` field of the resulting block. Only parameters provided by the user are included — optional fields that are omitted in the call MUST NOT appear in `tx`.
+
+All fields are encoded using the ICRC-3 `Value` type.
+
+---
+
+#### `icrc1_transfer`
 
-Example with `btype`:
+**Call parameters:**
+
+```candid
+icrc1_transfer: record {
+  to: Account;
+  amount: Nat;
+  fee: opt Nat;
+  memo: opt Blob;
+  from_subaccount: opt blob;
+  created_at_time: opt Nat;
+}
 ```
-variant { Map = vec {
-    record { "btype"; "variant" { Text = "1mint" }};
-    record { "ts"; variant { Nat = 1_675_241_149_669_614_928 : nat } };
-    record { "tx"; variant { Map = vec {
-        record { "amt"; variant { Nat = 100_000 : nat } };
-        record { "to"; variant { Array = vec {
-                variant { Blob = blob "Z\d0\ea\e8;\04*\c2CY\8b\delN\ea>]\ff\12^. WGj0\10\e4\02" };
-        }}};
-    }}};
-}};
+
+**Regular Transfer** — when neither the sender nor recipient is the minting account:
+
+- `op = "xfer"`
+- `from = [caller]` if `from_subaccount` is not provided  
+- `from = [caller, from_subaccount]` if provided
+- `to = to`
+- `amt = amount`
+- `fee = fee` if provided
+- `memo = memo` if provided
+- `ts = created_at_time` if provided
+
+
+
+
+**Transfer from the Minting Account (→ Mint)** — when `[caller]` or `[caller, from_subaccount]` equals the minting account:
+
+- `op = "mint"`
+- `to = to`
+- `amt = amount`
+- `memo = memo` if provided
+- `ts = created_at_time` if provided  
+- `from` and `fee` MUST NOT be present
+
+**Transfer to the Minting Account (→ Burn)** — when `to` equals the minting account:
```

Comment:
What about when both `caller` and `to` are the minting account? Similarly also for transfer from

Reply:
There's no good answer here, and although there are some restrictions in ICRC-2, this case is really underspecified. 
I suggest to simply prevent it explicitly. WDYT?